src/inference/graph_inference.cpp

```diff
@@ -341,6 +342,24 @@ class GraphNodeAssignment : public Nice2Assignment {
     return label_set_->GetLabelName(label_id);
   }
 
+  // Returns conditional probability of a label conditionated on its neighbouring labels
```

Comment:
Now I understand which functions you were referring to (GetNodeConditionalProbabilityGivenNeighbours and GetNodeNormalizationConstantGivenNeighbours). They are small and I can move them inside PLLearn. I agree with you they are better inside PLLearn.